File: AppFinancieraIOs-Development/BackendAPI/Models/Income.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BackendAPI.Models
{
    public class Income
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();
        
        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal GrossAmount { get; set; }
        
        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal NetAmount { get; set; }
        
        [Required]
        public DateTime Date { get; set; }
        
        [Required]
        public IncomeType Type { get; set; }
        
        [Required]
        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;
        
        [Required]
        public bool IsRecurring { get; set; }
        
        public RecurringPeriod? RecurringPeriod { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
    
    public enum IncomeType
    {
        Freelance = 0,
        Employment = 1,
        Investment = 2,
        Other = 3
    }
    
    public enum RecurringPeriod
    {
        Weekly = 0,
        Biweekly = 1,
        Monthly = 2,
        Yearly = 3
    }
}

File: AppFinancieraIOs-Development/Appfinancierafuncional/Appfinancierafuncional/Models/Income.swift
import Foundation

struct Income: Identifiable, Codable {
    var id: UUID
    var grossAmount: Double
    var netAmount: Double
    var date: Date
    var type: IncomeType
    var description: String
    var isRecurring: Bool
    var recurringPeriod: RecurringPeriod?
    
    // Inicializador personalizado para permitir ID opcional
    init(
        id: UUID = UUID(),
        grossAmount: Double,
        netAmount: Double,
        date: Date = Date(),
        type: IncomeType,
        description: String,
        isRecurring: Bool = false,
        recurringPeriod: RecurringPeriod? = nil
    ) {
        self.id = id
        self.grossAmount = grossAmount
        self.netAmount = netAmount
        self.date = date
        self.type = type
        self.description = description
        self.isRecurring = isRecurring
        self.recurringPeriod = recurringPeriod
    }
    
    enum IncomeType: String, CaseIterable, Codable {
        case freelance = "Freelance"
        case employment = "Empleo"
        case investment = "Inversión"
        case other = "Otro"
    }
    
    enum RecurringPeriod: String, CaseIterable, Codable {
        case weekly = "Semanal"
        case biweekly = "Quincenal"
        case monthly = "Mensual"
        case yearly = "Anual"
    }
    
    // Custom coding para mapear campos del backend (camelCase)
    enum CodingKeys: String, CodingKey {
        case id
        case grossAmount
        case netAmount
        case date
        case type
        case description
        case isRecurring
        case recurringPeriod
    }
}

